common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -186,7 +180,7 @@ public void onComplete(String streamId) throws IOException {
       ManagedBuffer buffer = new FileSegmentManagedBuffer(transportConf, targetFile, 0,
         targetFile.length());
       listener.onBlockFetchSuccess(blockIds[chunkIndex], buffer);
-      if (canCallerSideDeleteFile.get()) {
+      if (!tempShuffleFileManager.registerTempShuffleFileToClean(targetFile)) {
```

Comment:
Good catch! Previously we only put `FileSegmentManagedBuffer` in the `result` queue of `ShuffleBlockFetcherIterator`, and in `ShuffleBlockFetcherIterator.cleanup`, we forget to handle the temp file in the `result` queue.